projects/ng-polymorpheus/src/directives/outlet.ts

```diff
@@ -62,47 +73,34 @@ export class PolymorpheusOutletDirective<C> implements OnChanges, DoCheck {
             this.process(this.content, proxy);
         } else if (
             // tslint:disable-next-line:triple-equals
-            (context instanceof PolymorpheusContext && context.$implicit) != null
+            (context instanceof PolymorpheusContext && context.$implicit) !== null
         ) {
-            this.vcr.createEmbeddedView(this.template, proxy);
+            this.vcr.createEmbeddedView(this.template, proxy, {injector: this.i});
         }
     }
 
-    ngDoCheck() {
+    ngDoCheck(): void {
         if (isDirective(this.content)) {
             this.content.check();
         }
     }
 
-    static ngTemplateContextGuard<T>(
-        _dir: PolymorpheusOutletDirective<T>,
-        _ctx: any,
-    ): _ctx is PolymorpheusContext<T extends PolymorpheusPrimitive ? T : never> {
-        return true;
-    }
-
-    private getContext(): C | undefined | PolymorpheusContext<any> {
+    private getContext(): C | PolymorpheusContext<any> | undefined {
         if (isTemplate(this.content) || isComponent(this.content)) {
             return this.context;
         }
 
         return new PolymorpheusContext(
-            typeof this.content === 'function'
-                ? this.content(this.context!)
+            this.context && typeof this.content === 'function'
+                ? this.content(this.context)
                 : this.content,
         );
     }
 
     private process(content: PolymorpheusComponent<unknown>, proxy?: C): void {
         const injector = content.createInjector(this.i, proxy);
 
-        this.c = this.vcr.createComponent(
-            injector
-                .get(ComponentFactoryResolver)
-                .resolveComponentFactory(content.component),
-            0,
-            injector,
-        );
+        this.c = this.vcr.createComponent(content.component, {index: 0, injector});
```

Comment:
I don't think index is relevant, it was explicitly stated because injector was a third argument. If we need to keep it, we should also add it to template instantiation.